### ISSUE TITLE
Thread.cpp/.hpp: m_threadID must be cast in pthread_cancel

### ISSUE THREAD

**Issue by phillipjohnston** [johnttaylor/pim#7]
`pthread_self()` returns a `pthread_t`, which is a pointer. This is then cast to a `size_t` (aside: why not `uintptr_t`?). It is then used in `pthread_cancel` without casting back to the appropriate pointer type, which causes an error with my compiler:

```
/Users/phillip/src/ea/pim/src/Cpl/System/Posix/Thread.cpp:173:13: error: no matching function for call to 'pthread_cancel'
            pthread_cancel( m_threadID );
            ^~~~~~~~~~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/pthread.h:290:5: note: candidate function not viable: no known conversion from 'size_t' (aka 'unsigned long') to 'pthread_t _Nonnull' (aka '_opaque_pthread_t *') for 1st argument
int pthread_cancel(pthread_t) __DARWIN_ALIAS(pthread_cancel);
```

Again, I know you are targeting Linux and GCC, but the point the Clang is making here is valid - the `size_t` type requires a cast to match the API expectation of `pthread_cancel`

**Comment by johnttaylor** [comment]
I will provide a fix/change to provide the correct cast to the pthread_cancel().  The m_threadID member variable is of type size_t because that is what the non-platform specific Thread API defines a thread identifier to be.

**Comment by johnttaylor** [comment]
This has been resolved in commit [22c1e68](https://github.com/johnttaylor/pim/commit/22c1e68c8045d3fd5dad0f1ee023ec5d0abc67eb)
